### ISSUE TITLE
Feature : Checking if the auth token has expired 

### ISSUE THREAD

**Issue by epicadk** [anitab-org/bridge-in-tech-backend#206]
### Is your feature request related to a problem? Please describe.

Currently, we send all auth tokens to the MS-backend. However if we check if the auth token has expired or not before sending it to the MS-backend then it would reduce the load and also limit our calls to the MS-backend. 

### Describe the solution you'd like

Check if the Auth token has expired. And if it has send a 401 response to the user. 

### Additional context

There is an open pr for tsomething similar on the MS-android repo as well.

**Comment by epicadk** [comment]
Marking this as status on hold as this needs to be approved by @mtreacy002.

**Comment by mtreacy002** [comment]
@epicadk , can you please share the link to the similar issue on the MS Android so I can check it out?

**Comment by epicadk** [comment]
Sure , [here](https://github.com/anitab-org/mentorship-android/pull/1023/files#diff-9d30af116c1a33ffe266b84261069652ef46833d700be8cc18d128c19427e932R50) it is. Although that is only a small part of the pr. I Think it is on line 50 in the network interceptor file :) .

**Comment by mtreacy002** [comment]
I'm thinking if we could do the same so instead of checking the token expiry on BIT backend, we can do that on BIT web, that way we don't need to send request from frontend if token is expired. We just need to ask user to login again on frontend. What do you think, @epicadk ?

**Comment by epicadk** [comment]
> I'm thinking if we could do the same so instead of checking the token expiry on BIT backend, we can do that on BIT web, that way we don't need to send request from frontend if token is expired. We just need to ask user to login again on frontend. What do you think, @epicadk ?

I agree but since it is a rest api and a user does not necessarily need to use the BIT front-end I still think it makes sense to add a check to the BIT backend.

**Comment by mtreacy002** [comment]
That makes sense, it'll be good for the backend developers so it reminds them that token is expired meaning they need to login again 👍. Ok. will approve this issue now. Would you like to work on this, @epicadk ?

**Comment by epicadk** [comment]
> That makes sense, it'll be good for the backend developers so it reminds them that token is expired meaning they need to login again 👍. Ok. will approve this issue now. Would you like to work on this, @epicadk ?

Yep, I'd like to work on this. : )

**Comment by epicadk** [comment]
Although it might take a while because I have to set the project up locally. I hope that's okay : ).

**Comment by mtreacy002** [comment]
Assigning you now 😉. Take your time. Just don't forget to give update every 3-4 days.

**Comment by epicadk** [comment]
Making this issue available as I have to work on other action items.

**Comment by Rahulm2310** [comment]
@epicadk @mtreacy002 I would like to try this 🙂

**Comment by epicadk** [comment]
> @epicadk @mtreacy002 I would like to try this 🙂

Assigning you good luck. 👍🏼

**Comment by mtreacy002** [comment]
@Rahulm2310 , you're currently assigned to BIT web issue [#155](https://github.com/anitab-org/bridge-in-tech-web/issues/155) and haven't submitted a PR for  a review. Can you please focus on that one first? cc @epicadk

**Comment by epicadk** [comment]
> @Rahulm2310 , you're currently assigned to BIT web issue [#155](https://github.com/anitab-org/bridge-in-tech-web/issues/155) and haven't submitted a PR for a review. Can you please focus on that one first? cc @epicadk

Oh I thought it was one issue per repository. I'll keep this in mind next time. : )
